qa/tasks/rgw.py

```diff
@@ -1029,26 +943,35 @@ def pull_configuration(ctx, config, regions, role_endpoints, realm, master_clien
 
     for client in config.iterkeys():
         if client != master_client:
+            cluster_name, daemon_type, client_id = teuthology.split_role(client)
             host, port = role_endpoints[master_client]
             endpoint = 'http://{host}:{port}/'.format(host=host, port=port)
             log.debug("endpoint: %s", endpoint)
             rgwadmin(ctx, client,
-                cmd=['-n', client, 'realm', 'pull', '--rgw-realm', realm, '--default', '--url',
+                cmd=['realm', 'pull', '--rgw-realm', realm, '--default', '--url',
                      endpoint, '--access_key',
                      user_info['system_key']['access_key'], '--secret',
                      user_info['system_key']['secret_key']],
                      check_status=True)
 
             (zonegroup, zone, zone_info, zone_user_info) = role_zones[client]
             zone_json = json.dumps(dict(zone_info.items() + zone_user_info.items()))
-            log.debug("zone info is: %r"), zone_json
+            log.debug("zone info is: %r", zone_json)
             rgwadmin(ctx, client,
-                     cmd=['zone', 'set', '--rgw-zonegroup', zonegroup,
+                     cmd=['zone', 'set', '--default',
```

Comment:
great questions. @cbodley do you remember? I remember there was an issue and this fix solved it, but I can't remember more than that.